### ISSUE TITLE
Invalid JS: Splatted statement causes if block inside array element

### ISSUE THREAD

**Issue by satyr** [jashkenas/coffeescript#1349]
```
$ coffee -bpe '[a if b...]'
var __slice = Array.prototype.slice;
__slice.call(  if (b) {
    a;
  });

$ coffee -e '[a if b...]'

.:3
  __slice.call(    if (b) {
                   ^^
SyntaxError: Unexpected token if
```

**Comment by GeoffreyBooth** [comment]
Either a duplicate or else very closely related to #4260.

**Comment by GeoffreyBooth** [comment]
Fixed by #4644.
